app/Platform/Controllers/BeatenGamesLeaderboardController.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Controllers;
+
+use App\Community\Enums\AwardType;
+use App\Http\Controller;
+use App\Platform\Enums\UnlockMode;
+use App\Platform\Models\System;
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\Request;
+use Illuminate\Pagination\LengthAwarePaginator;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+
+class BeatenGamesLeaderboardController extends Controller
+{
+    private int $pageSize = 25;
+
+    public function __invoke(Request $request): View
+    {
+        if (!config('feature.beat')) {
+            abort(404);
+        }
+
+        $validatedData = $request->validate([
+            'page.number' => 'sometimes|integer|min:1',
+            'filter.system' => 'sometimes|integer',
+            'filter.retail' => 'sometimes|in:true,false',
+            'filter.hacks' => 'sometimes|in:true,false',
+            'filter.homebrew' => 'sometimes|in:true,false',
+            'filter.unlicensed' => 'sometimes|in:true,false',
+            'filter.prototypes' => 'sometimes|in:true,false',
+            'filter.demos' => 'sometimes|in:true,false',
+        ]);
+
+        $targetSystemId = (int) ($validatedData['filter']['system'] ?? 0);
+
+        $gameKindFilterOptions = [
+            'retail' => ($validatedData['filter']['retail'] ?? true) !== 'false',
+            'hacks' => ($validatedData['filter']['hacks'] ?? true) !== 'false',
+            'homebrew' => ($validatedData['filter']['homebrew'] ?? true) !== 'false',
+            'unlicensed' => ($validatedData['filter']['unlicensed'] ?? true) !== 'false',
+            'prototypes' => ($validatedData['filter']['prototypes'] ?? true) !== 'false',
+            'demos' => ($validatedData['filter']['demos'] ?? true) !== 'false',
+        ];
+
+        // We need to know how many rows there are, otherwise the
+        // paginator can't determine what the max page number should be.
+        $rankedRowsCount = $this->getLeaderboardRowCount($targetSystemId, $gameKindFilterOptions);
+
+        // Now get the current page's rows.
+        $currentPage = $validatedData['page']['number'] ?? 1;
+        $offset = (int) ($currentPage - 1) * $this->pageSize;
+
+        $beatenGameAwardsRankedRows = $this->getLeaderboardDataForCurrentPage($offset, $targetSystemId, $gameKindFilterOptions);
```

Comment:
The sort seems to be alphabetical by user now. This is page 1:
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/fd6db1e3-3d9b-44b0-a910-35faec795f41)

Reply:
Could this be a side effect of using MariaDB? I am definitely not seeing this.

![Screenshot 2023-09-13 at 10 08 39 AM](https://github.com/RetroAchievements/RAWeb/assets/3984985/4efd43e9-d593-4eb6-adaa-2fd4f3e97d1b)

@Tsearo Can you repro this issue?

Reply:
Rank ordering worked fine for me during testing (MySQL) so it's possible it could be a MariaDB issue

Reply:
The test passes. Here's the query that's being run:
```
select * from (select `User`, 
    RANK() OVER (ORDER BY COUNT(s.User) DESC) as rank_number,
    ROW_NUMBER() OVER (ORDER BY COUNT(s.User) DESC) as leaderboard_row_number,
    COUNT(s.User) as total_awards, `most_recent_game_id`, `Title` as `GameTitle`,
    `ImageIcon` as `GameIcon`, `ConsoleName`, 
    `AwardDate` as `last_beaten_date` from (
        select `ua`.`User` as `User`,
            FIRST_VALUE(gd.ID) OVER (PARTITION BY ua.User ORDER BY sa.AwardDate desc) as most_recent_game_id,
            `gd`.`Title`, `gd`.`ImageIcon`, `c`.`name` as `ConsoleName`, `sa`.`AwardDate`
            from `SiteAwards` as `sa` inner join `GameData` as `gd` on `sa`.`AwardData` = `gd`.`ID`
            inner join `Console` as `c` on `gd`.`ConsoleID` = `c`.`ID`
            inner join `UserAccounts` as `ua` on `sa`.`User` = `ua`.`User`
            where `sa`.`AwardType` = 8 and `sa`.`AwardDataExtra` = 1 and `ua`.`Untracked` = 0
            and `gd`.`Title` not like '~Subset~%' and `gd`.`Title` not like '~Test Kit~%'
            and `gd`.`Title` not like '~Multi~%' and `gd`.`ConsoleID` not in (100, 101)
    ) as s
    group by `User`
) as b limit 25 offset 0
```
There probably needs to be an outer-level ORDER BY.

Reply:
Outer-level `ORDER BY` added in https://github.com/RetroAchievements/RAWeb/pull/1862/commits/38a30ffa76b61f7f2e0da092e868e649ad0ea141.